service/storage_service.cc

```diff
@@ -3270,10 +3267,10 @@ future<token_metadata_change> storage_service::prepare_token_metadata_change(mut
         });
         // Prepare per-table erms.
         co_await container().invoke_on_all([&] (storage_service& ss) -> future<> {
-            auto& db = ss._db.local();
             auto tmptr = change.pending_token_metadata_ptr[this_shard_id()];
-            co_await db.get_tables_metadata().for_each_table_gently([&] (table_id id, lw_shared_ptr<replica::table> table) {
-                auto rs = db.find_keyspace(table->schema()->ks_name()).get_replication_strategy_ptr();
+            auto keyspaces = schema_getter.get_keyspaces();
+            co_await schema_getter.for_each_table_gently([&] (table_id id, lw_shared_ptr<replica::table> table) {
```

Comment:
bisectability: There is no schema_getter::for_each_table_gently() in this commit.